Shared/Samples/Augment reality to show hidden infrastructure/AugmentRealityToShowHiddenInfrastructureView.swift

```diff
@@ -0,0 +1,262 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import CoreLocation
+import SwiftUI
+
+struct AugmentRealityToShowHiddenInfrastructureView: View {
+    /// The view model for the map view in the sample.
+    @StateObject private var model = MapModel()
+    
+    /// The status message in the overlay.
+    @State private var statusMessage = "Tap the map to add pipe points."
+    
+    /// A Boolean value indicating whether there are graphics to be deleted.
+    @State private var canDelete = false
+    
+    /// A Boolean value indicating whether the current geometry edits can be added as a pipe.
+    @State private var canApplyEdits = false
+    
+    /// A Boolean value indicating whether the geometry editor can undo.
+    @State private var geometryEditorCanUndo = false
+    
+    /// A Boolean value indicating whether the alert for entering an elevation offset is showing.
+    @State private var elevationAlertIsPresented = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        NavigationView {
+            MapView(map: model.map, graphicsOverlays: [model.pipesGraphicsOverlay])
+                .locationDisplay(model.locationDisplay)
+                .geometryEditor(model.geometryEditor)
+                .toolbar {
+                    ToolbarItemGroup(placement: .bottomBar) {
+                        toolbarButtons
+                    }
+                }
+        }
+        .overlay(alignment: .top) {
+            instructionText
+        }
+        .elevationOffsetAlert(isPresented: $elevationAlertIsPresented) { elevationOffset in
+            model.addPipe(elevationOffset: elevationOffset)
+            canDelete = true
+            
+            if elevationOffset < 0 {
+                statusMessage = "Pipe added \(elevationOffset.formatted()) meter(s) below surface."
+            } else if elevationOffset == 0 {
+                statusMessage = "Pipe added at ground level."
+            } else {
+                statusMessage = "Pipe added \(elevationOffset.formatted()) meter(s) above surface."
+            }
+            statusMessage.append("\nTap the camera to view the pipe(s) in AR.")
+            
+            model.geometryEditor.start(withType: Polyline.self)
+        }
+        .task {
+            do {
+                try await model.startLocationDisplay()
+            } catch {
+                self.error = error
+            }
+            
+            // Start the geometry editor and listen for its geometry updates.
+            model.geometryEditor.start(withType: Polyline.self)
+            
+            for await geometry in model.geometryEditor.$geometry {
+                let polyline = geometry as? Polyline
+                canApplyEdits = polyline?.parts.contains { $0.points.count >= 2 } ?? false
```

Comment:
This will show instructions to tap "Done" before the alert is presented.
<img width="20%" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/16397058/c3f4531e-8197-4f77-bb25-3868f83c4dac">

Maybe "Tap the check to add the pipe" instead? Or consider changing the check to a "Done" button instead.